### ISSUE TITLE
Better validation of YAML?

### ISSUE THREAD

**Issue by shimonp21** [cloudquery/cloudquery#1104]
### Describe the Bug

Resources are fetched from all regions, not the regions specified in config.

### Expected Behavior
resources should only be fetched from the resources specified in config

### CloudQuery Version
Version: 0.30.0
provider version: v0.12.25 

### Steps to Reproduce
```
cloudquery:
    providers:
        - name: aws
          version: latest
    connection:
        type: postgres
        username: postgres
        password: pass
        host: localhost
        port: 5432
        database: postgres
        sslmode: disable
providers:
    # provider configurations
    - name: aws
      configuration:
      # Optional, Repeated. Add an accounts block for every account you want to assume-role into and fetch data from.
      # accounts:
      #   - id: <UNIQUE ACCOUNT IDENTIFIER>
      # Optional. Role ARN we want to assume when accessing this account
      #     role_arn: < YOUR_ROLE_ARN >
      # Optional. Named profile in config or credential file from where CQ should grab credentials
      #     local_profile: < PROFILE_NAME >
      # Optional. by default assumes all regions
      regions:
        - eu-west-3
      # Optional. Enable AWS SDK debug logging.
      #   aws_debug: false
      # The maximum number of times that a request will be retried for failures. Defaults to 10 retry attempts.
      # max_retries: 10
      # The maximum back off delay between attempts. The backoff delays exponentially with a jitter based on the number of attempts. Defaults to 30 seconds.
      # max_backoff: 30
      #
      # list of resources to fetch
      resources:
        - ec2.vpcs
        - ec2.instances
```

**Comment by disq** [comment]
You are missing whitespace, `regions` should be indented to be inside `configuration` block, which it's currently not. Blame ~Python~ YAML

**Comment by shimonp21** [comment]
Then I guess the action-item here is better validation?

**Comment by bbernays** [comment]
Initial improvements were merged in #1106 . Definitely more work to be done on the provider side of things

**Comment by bbernays** [comment]
Can we add a validation command to the SDK so we can achieve the same results on all of the providers?
